### PR TITLE
Refactor OAM handler and add metadata for new Meshery UI

### PR DESCRIPTION
**Pull request by utkarsh-pro**
**Description**

This PR:
1. Refactors the OAM handler code to make it more generic
2. Adds some metadata which is used in the UI (the metadata except the adapter service mesh name is not required for the UI to work properly)

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: istio/oam.go [hunk 1]
```diff
@@ -11,46 +11,33 @@ import (
...
-			if err := handleComponentVirtualService(istio, comp, isDel); err != nil {
-				errs = append(errs, err)
-			}
 
-			msg := fmt.Sprintf("created virtual service \"%s\" in namespace \"%s\"", comp.Name, comp.Namespace)
-			if isDel {
-				msg = fmt.Sprintf("deleted virtual service \"%s\" in namespace \"%s\"", comp.Name, comp.Namespace)
-			}
+	compFuncMap := map[string]CompHandler{
+		"IstioMesh":            handleComponentIstioMesh,
```
Comment (kumarabd) [inline istio/oam.go:23]: @utkarsh-pro Can this be moved to `config/`? Just the declarations.

File: istio/oam/register.go [hunk 2]
```diff
@@ -46,10 +47,19 @@ func RegisterWorkloads(runtime, host string) error {
 	for _, workload := range workloads {
 		defintionPath, schemaPath := generatePaths(workloadPath, workload)
 
+		metadata := map[string]string{
+			"adapter.meshery.io/name": "istio",
+		}
```
Comment (kumarabd) [inline istio/oam/register.go:52]: This needs to be a part of `config/`

File: istio/oam/register.go [hunk 3]
```diff
@@ -77,6 +87,9 @@ func RegisterTraits(runtime, host string) error {
 			OAMDefintionPath: defintionPath,
 			OAMRefSchemaPath: schemaPath,
 			Host:             host,
+			Metadata: map[string]string{
```
Comment (kumarabd) [inline istio/oam/register.go:90]: Reuse the above from the config

File: istio/oam.go (hunk 1 above)
Comment (utkarsh-pro) [reply istio/oam.go:23]: Won't that create circular dependency issue? This map's values are functions of type `func(*Istio, v1alpha1.Component, bool) (string, error)`, so it needs `Istio` from this package.

**Comment by utkarsh-pro** [comment]
@kumarabd does this recent change addresses your comments?

File: istio/oam.go (hunk 1 above)
Comment (kumarabd) [reply istio/oam.go:23]: makes sense. On a different note, are the contents inside `oam/` common across adapters? If not we can move this into that and import. Basically to manage this list independently in a file.

**Comment by kumarabd** [comment]
> @kumarabd does this recent change addresses your comments?

yes it does, Thank you.

File: istio/oam.go (hunk 1 above)
Comment (utkarsh-pro) [reply istio/oam.go:23]: Actually almost everything that was generic here has been already moved to `meshery-adapter-library` :sweat_smile:

**Comment by utkarsh-pro** [comment]
Can we move this PR ahead? Actually it is blocking the EnvoyFilter PR :sweat_smile:

**Pull request merged**